### DISCUSSION TITLE
Receiving fork/exec gauge-java: exec format error

### DISCUSSION THREAD

**Discussion by shermanericts** [getgauge/gauge#2207] [General]
Hi there.

After running 
```
sudo curl -fsSL https://downloads.gauge.org/stable | sh
gauge install java
mvn gauge:execute
```

I'm seeing this error:
```
INFO]
[INFO] --- gauge-maven-plugin:1.5.0:execute (default-cli) @ ingress ---
Error ----------------------------------

[Gauge]
Failed to start gauge API: Error occurred while starting runner process.
Error : fork/exec gauge-java: exec format error
```

Specifics:
```
Your Environment Information -----------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
	linux, 1.4.3, f98dd40
[INFO] Total time:  3.067 s
	html-report (4.1.4), java (0.8.0), screenshot (0.1.0)
[INFO] Finished at: 2022-02-21T05:10:09Z
```

I've tried this on the following Operating Systems:
```
debian-11-bullseye-v20211105
ubuntu-2004-focal-v20220204 (lts)
```

Happy to try to give more debugging information.  This had all been working up until roughly 2 days ago.  I'm running these on VMs in our Google Compute Engine environments. 

Thanks in advance!

**Comment by thjenny** [comment]
to me it looks like a packaging error for gauge-java 0.8.0.

Downloading this: https://github.com/getgauge/gauge-java/releases/download/v0.8.0/gauge-java-0.8.0-linux**.x86_64.**zip
Unpacking, then 
file gauge-java

gauge-java: ELF 64-bit LSB executable, **ARM aarch64**, version 1 (SYSV), statically linked, Go BuildID=aKs9Pl6rxYX7AB0GhGWR/8cMA-eWRziIeq11tQ3VZ/QCAZMY174-necWe0yUkw/AdjD51yRzMWx7WFxB-Pz, stripped

Not what I'd expect in the ZIP file named gauge-java-0.8.0-linux.x86_64.zip

**Comment by zabil** [comment]
Is there any other info that might help to replicate this?

This works on Java 17, Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537) and 

```
Gauge version: 1.4.3
Plugins
-------
html-report (4.1.4)
java (0.8.0)
screenshot (0.1.0)
```
```
gauge init java_maven
mvn -q compile 
mvn -q gauge:execute
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
# Specification Heading
  ## Vowel counts in single word         ✔ ✔
  ## Vowel counts in multiple word       ✔ ✔

Successfully generated html-report to => /Users/zabilcm/Gauge/vowels/reports/html-report/index.html

Specifications: 1 executed      1 passed        0 failed        0 skipped
Scenarios:      2 executed      2 passed        0 failed        0 skipped

Total time taken: 83ms
```

**Comment by thjenny** [comment]
the issue is a packaging issue.
The zip file for linux x86_64 does **not** contain a x86_64 binary of gauge-java, but instead the **ARM aarch64** binary.

**Reply by zabil** [reply]
Thanks for pointing that out. cc/ @sriv

**Comment by sriv** [comment]
I am investigating this right away

**Reply by sriv** [reply]
Something strange happened on the github action that deployed - https://github.com/getgauge/gauge-java/runs/5252435254?check_suite_focus=true#step:7:104

On my local this looks like:

```
Compiling for platform => OS:darwin ARCH:arm64 
2022/02/21 14:47:08 Execute [go build -ldflags=-s -w -o bin/darwin_arm64]
Compiling for platform => OS:darwin ARCH:amd64 
2022/02/21 14:47:09 Execute [go build -ldflags=-s -w -o bin/darwin_amd64]
Compiling for platform => OS:linux ARCH:386 
2022/02/21 14:47:09 Execute [go build -ldflags=-s -w -o bin/linux_386]
Compiling for platform => OS:linux ARCH:amd64 
2022/02/21 14:47:09 Execute [go build -ldflags=-s -w -o bin/linux_amd64]
Compiling for platform => OS:linux ARCH:arm64 
2022/02/21 14:47:09 Execute [go build -ldflags=-s -w -o bin/linux_arm64]
Compiling for platform => OS:windows ARCH:386 
2022/02/21 14:47:09 Execute [go build -ldflags=-s -w -o bin/windows_386]
Compiling for platform => OS:windows ARCH:amd64 
2022/02/21 14:47:09 Execute [go build -ldflags=-s -w -o bin/windows_amd64]
```

On the CI, for some reason the env var and `go build` execution are out of sync. Not sure if it is just the logging. But that does explain the mixup of arm64 binary on x86_64 artifact

**Reply by sriv** [reply]
I tried a re-deploy, and that fails for the same reason. I am going to take out the arm64 builds for linux for now. Will add it in later if we needed.

**Reply by thjenny** [reply]
does that mean it is supposed to be fixed? 
Because if I download the x86_64 linux version now, I still get the ARM binary.
I also tried with wget --no-cache, same result, still ARM aarch64.

**Reply by sriv** [reply]
this continues to be an issue and needs to be fixed.

**Reply by sriv** [reply]
raised PR https://github.com/getgauge/gauge-java/pull/673 which should fix this.

**Reply by sriv** [reply]
@thjenny @shermanericts  - could you please try installing the latest gauge-java (0.8.1) this problem should be fixed.

**Reply by thjenny** [reply]
thanks for the quick fix.
When I download the plugin, and install manually (`gauge install java -f ./gauge-java-0.8.1-linux.x86_64.zip`), it does work fine.
Automatic update does not (yet?) work because https://downloads.gauge.org/plugin/java is not yet updated.

**Reply by sriv** [reply]
Ah yes, it takes a while for the cache to get updated. It should be ready in a few hours time

**Reply by shermanericts** [reply]
I'll try this shortly. thanks @sriv  for the quick turnaround. I'll let you know!

**Reply by thjenny** [reply]
the cache is up to date now, and - as expected - it also works now again with automatically updated plugin.
Thanks again.

**Reply by shermanericts** [reply]
thanks @thjenny .  Local testing looked good to me too!

**Reply by sriv** [reply]
Thanks for your patience and confirmation @thjenny @shermanericts
